flexmeasures/utils/coding_utils.py

```diff
@@ -69,19 +69,29 @@ def sort_dict(unsorted_dict: dict) -> dict:
     return sorted_dict
 
 
+# Modified flatten_unique to support new sensor format with titles
 def flatten_unique(nested_list_of_objects: list) -> list:
     """Returns unique objects in a possibly nested (one level) list of objects.
 
     Preserves the original order in which unique objects first occurred.
 
+    Handles lists, individual sensor IDs, and dictionaries with sensor titles.
+
     For example:
-    >>> flatten_unique([1, [2, 20, 6], 10, [6, 2]])
-    <<< [1, 2, 20, 6, 10]
+    >>> flatten_unique([1, [2, 20, 6], 10, {"title": "row1", "sensor": [6, 2]}, {"title": "row2", "sensor": 7}])
+    <<< [1, 2, 20, 6, 10, 7]
     """
     all_objects = []
     for s in nested_list_of_objects:
         if isinstance(s, list):
             all_objects.extend(s)
+        elif isinstance(s, dict):
+            # Handle dictionary format
+            sensors = s["sensor"]
```

Comment:
Could also be `s["sensors"]`, I think. At least, that's what the `sensors_to_show` method's docstring currently suggests (which I liked). Let's add a test case. Is there a test already for this util?

Reply:
@Flix6x Do you mean to change `{'title': 'sensor_title', 'sensor': sensor_id}` to `{'title': 'sensor_title', 'sensors': sensor_id}` in `sensor_to_show`? It seems I mistakenly used `'sensor'` and `'sensors'`  in the docstring exmple. However, the logic of the changes I made is based on `'sensor'`.

I also didn't find a test in utils for this. I've been manually testing various cases and they provide correct responses, But i can add a test case.

Reply:
Thank you both for resolving the clarification together. And yes, please add some test cases for the schema.